needy/filesystem.py

```diff
@@ -48,9 +82,30 @@ def lock_file(file_path, timeout=None):
                 fcntl.flock(fd, fcntl.LOCK_EX)
         else:
             fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-        return fd
     except IOError:
-        pass
+        return False
+    return True
+
+
+def lock_fd(fd, timeout=None):
```

Comment:
There seem to be a lot of places where it's more convenient or even necessary to lock a file-like object. So this method lets you do that by passing the file-like object's `fileno()`.